### PR TITLE
KeyAccessSample refactoring

### PR DESCRIPTION
**Pull request by kitaisreal**
KeyAccessSample refactoring

### CODE REVIEW

File: ydb/core/tablet_flat/flat_stat_table.h [hunk 1]
```diff
@@ -140,40 +140,38 @@ class TKeyAccessSample {
...
         if (idx >= SampleCount) {
             return;
         }
 
-        TSerializedCellVec saved(TSerializedCellVec::Serialize(key));
+        TString serializedKey = TSerializedCellVec::Serialize(key);
 
-        auto it = KeyRefCount.find(saved.GetBuffer());
-        if (it != KeyRefCount.end()) {
+        auto [it, inserted] = KeyRefCount.emplace(serializedKey, 1);
```
Comment (snaury) [inline ydb/core/tablet_flat/flat_stat_table.h:152]: I thinks this should be replaced with `++KeyRefCount[serializedKey]`. The `it` iterator is not used for anything else anyway.

File: ydb/core/tablet_flat/flat_stat_table.h [hunk 2]
```diff
@@ -140,40 +140,38 @@ class TKeyAccessSample {
...
+            return;
+        }
+
+        TString old = Sample[idx].first;
+        auto oit = KeyRefCount.find(old);
+        Y_VERIFY(oit != KeyRefCount.end());
 
-            Sample[idx] = std::make_pair(saved.GetBuffer(), accessKind);
+        // Delete the key if this was the last reference
+        if (oit->second == 1) {
```
Comment (snaury) [inline ydb/core/tablet_flat/flat_stat_table.h:168]: Could be simplified to `if (0 == --oit->second) { KeyRefCount.erase(oit); }`. Comparions with `0` usually have more mechanical sympathy.

**Review by snaury — CHANGES_REQUESTED** [review]

File: ydb/core/tablet_flat/flat_stat_table.h (hunk 1 above)
Comment (kitaisreal) [reply ydb/core/tablet_flat/flat_stat_table.h]: Updated.

File: ydb/core/tablet_flat/flat_stat_table.h (hunk 2 above)
Comment (kitaisreal) [reply ydb/core/tablet_flat/flat_stat_table.h]: Updated.

**Review by snaury — APPROVED** [review]
LGTM

**Pull request closed without merging**